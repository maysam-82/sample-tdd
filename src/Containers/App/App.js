import React, { Component } from 'react';
import { Button } from 'react-bootstrap';
import Item from '../Item/Item';
import { maxNumber } from '../../Helper';

class App extends Component {
	constructor() {
		super();
		this.state = {
			items: [],
		};
	}
	onAddItem = () => {
		const { items } = { ...this.state };
		const ids = items.map(({ id }) => id);
		items.push({ id: maxNumber(ids) + 1 });
		this.setState({ items });
	};
	onRemoveItem = itemId => {
		const { items } = this.state;
		const filteredItemsWithoutItemId = items.filter(({ id }) => id !== itemId);
		this.setState({ items: filteredItemsWithoutItemId });
	};
	render() {
		const { items } = this.state;
		return (
			<div className="container">
				<header className="header">Sample Test Driven Development Project</header>
				<div className="items-list">
					{items && items.length
						? items.map(({ id }) => {
								return <Item key={id} itemId={id} onRemoveItem={this.onRemoveItem} />;
						  })
						: null}
				</div>
				<Button className="btn-add" onClick={this.onAddItem}>
					Add New Item
				</Button>
			</div>
		);
	}
}

export default App;
