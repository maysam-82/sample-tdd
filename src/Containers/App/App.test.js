import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App component', () => {
	const app = shallow(<App />);
	it('renders correctly', () => {
		expect(app.debug()).toMatchSnapshot();
	});
	it('initializes the `state` with an empty list of items', () => {
		expect(app.state().items).toEqual([]);
	});
	it('adds a new item to `state` items array while clicking the `add items` button', () => {
		app.find('.btn-add').simulate('click');
		expect(app.state().items).toEqual([{ id: 1 }]);
	});
});
