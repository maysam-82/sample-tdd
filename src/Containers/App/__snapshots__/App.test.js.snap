// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly 1`] = `
"<div className=\\"container\\">
  <header className=\\"header\\">
    Sample Test Driven Development Project
  </header>
  <Button className=\\"btn-add\\" onClick={[Function]} variant=\\"primary\\" active={false} disabled={false} type=\\"button\\">
    Add New Item
  </Button>
</div>"
`;
